examples/tv-app/android/java/MessagesManager.cpp

```diff
@@ -0,0 +1,425 @@
+
+            jfieldID getMessageIdField = env->GetFieldID(messageClass, "messageId", "Ljava/lang/String;");
+            jstring jmessageId         = static_cast<jstring>(env->GetObjectField(messageObject, getMessageIdField));
+            JniUtfString messageId(env, jmessageId);
+            if (jmessageId != nullptr)
+            {
+                VerifyOrReturnValue(chip::Encoding::HexToBytes(messageId.charSpan().data(), messageId.charSpan().size(), buf,
+                                                               sizeof(buf)) == sizeof(buf),
+                                    CHIP_ERROR_INVALID_ARGUMENT, ChipLogError(Zcl, "HexToBytes failed"));
+                message.messageID = ByteSpan(buf, sizeof(buf));
+            }
+
+            jfieldID getMessageTextField = env->GetFieldID(messageClass, "messageText", "Ljava/lang/String;");
+            jstring jmessageText         = static_cast<jstring>(env->GetObjectField(messageObject, getMessageTextField));
+            JniUtfString messageText(env, jmessageText);
+            if (jmessageText != nullptr)
+            {
+                message.messageText = messageText.charSpan();
+            }
+
+            jfieldID messageControlField = env->GetFieldID(messageClass, "messageControl", "I");
+            jint jmessageControl         = env->GetIntField(messageObject, messageControlField);
+            message.messageControl       = static_cast<chip::BitMask<MessageControlBitmap>>(static_cast<uint8_t>(jmessageControl));
+
+            jfieldID startTimeField = env->GetFieldID(messageClass, "startTime", "J");
+            jint jstartTime         = env->GetIntField(messageObject, startTimeField);
+            if (jstartTime >= 0)
+            {
+                message.startTime = DataModel::Nullable<uint32_t>(static_cast<uint32_t>(jstartTime));
+            }
+
+            jfieldID durationField = env->GetFieldID(messageClass, "duration", "I");
+            jint jduration         = env->GetIntField(messageObject, durationField);
+            if (jduration >= 0)
+            {
+                message.duration = DataModel::Nullable<uint16_t>(static_cast<uint16_t>(jduration));
+            }
+
+            jfieldID getResponseOptionsField = env->GetFieldID(messageClass, "responseOptions", "Ljava/lang/Vector;");
+
+            jobjectArray responsesArray = (jobjectArray) env->GetObjectField(messageObject, getResponseOptionsField);
+            jint size                   = env->GetArrayLength(responsesArray);
+            if (size > 0)
+            {
+                // MessageResponseOption * optionArray = new MessageResponseOption[static_cast<size_t>(size)];
+                // VerifyOrReturnError(optionArray != nullptr, CHIP_ERROR_NO_MEMORY,
+                //                     ChipLogProgress(Controller, "HandleGetMessages MessageResponseOption alloc failed"));
+                // optionToFree.push_back(optionArray); // TODO
```

Comment:
TODO for what? please create a ticket for follow-up, thanks

Reply:
removed